koku/api/report/gcp/serializers.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""GCP Report Serializers."""
+import logging
+
+from pint.errors import UndefinedUnitError
+from rest_framework import serializers
+
+from api.report.serializers import FilterSerializer as BaseFilterSerializer
+from api.report.serializers import GroupSerializer
+from api.report.serializers import OrderSerializer
+from api.report.serializers import ParamSerializer
+from api.report.serializers import StringOrListField
+from api.report.serializers import validate_field
+from api.utils import UnitConverter
+
+LOG = logging.getLogger(__name__)
+
+
+class GCPGroupBySerializer(GroupSerializer):
+    """Serializer for handling query parameter group_by."""
+
+    _opfields = ("account", "region", "service", "project")
+
+    account = StringOrListField(child=serializers.CharField(), required=False)
+    region = StringOrListField(child=serializers.CharField(), required=False)
+    service = StringOrListField(child=serializers.CharField(), required=False)
+    project = StringOrListField(child=serializers.CharField(), required=False)
+
+
+class GCPOrderBySerializer(OrderSerializer):
+    """Serializer for handling query parameter order_by."""
+
+    _opfields = ("account", "region", "service", "project")
+
+    account = serializers.ChoiceField(choices=OrderSerializer.ORDER_CHOICES, required=False)
+    region = serializers.ChoiceField(choices=OrderSerializer.ORDER_CHOICES, required=False)
+    service = serializers.ChoiceField(choices=OrderSerializer.ORDER_CHOICES, required=False)
+    project = serializers.ChoiceField(choices=OrderSerializer.ORDER_CHOICES, required=False)
+
+
+class GCPFilterSerializer(BaseFilterSerializer):
+    """Serializer for handling query parameter filter."""
+
+    _opfields = ("account", "service", "region", "project")
+
+    account = StringOrListField(child=serializers.CharField(), required=False)
+    service = StringOrListField(child=serializers.CharField(), required=False)
+    region = StringOrListField(child=serializers.CharField(), required=False)
+    project = StringOrListField(child=serializers.CharField(), required=False)
+
+
+class GCPQueryParamSerializer(ParamSerializer):
+    """Serializer for handling GCP query parameters."""
+
+    # Tuples are (key, display_name)
+    DELTA_CHOICES = ("cost", "cost")
```

Comment:
This needs to be a tuple of tuples. See, for example, koku/api/report/aws/serializers.py

```suggestion
    DELTA_CHOICES = (("cost", "cost"),)
```